huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go

```diff
@@ -0,0 +1,203 @@
+	if err != nil {
+		return diag.Errorf("error getting api information from server: %s", err)
+	}
+	if len(resp) < 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	return diag.FromErr(d.Set("publish_ids", flattenApiPublishIds(resp)))
+}
+
+func unbindPolicy(client *golangsdk.ServiceClient, instanceId, policyId string, unbindSet *schema.Set) error {
+	opt := buildListOpts(instanceId, policyId)
+	resp, err := throttles.ListBind(client, opt)
+	if err != nil {
+		return fmt.Errorf("error getting api information from server: %s", err)
+	}
+	for _, rm := range unbindSet.List() {
+		for _, api := range resp {
+			// If the publish ID is not found, it means the policy has been unbound from the API by other ways.
+			if rm == api.PublishId {
+				err = throttles.Unbind(client, instanceId, api.ThrottleApplyId)
+				if err != nil {
+					return fmt.Errorf("error unbound policy from the API: %s", err)
+				}
+				break
+			}
+		}
+	}
+	return nil
+}
+
+func resourceThrottlingPolicyAssociateUpdate(ctx context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ApigV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating HuaweiCloud APIG v2 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	policyId := d.Get("policy_id").(string)
+	oldRaw, newRaw := d.GetChange("publish_ids")
+	addSet := newRaw.(*schema.Set).Difference(oldRaw.(*schema.Set))
+	rmSet := oldRaw.(*schema.Set).Difference(newRaw.(*schema.Set))
+
+	if rmSet.Len() > 0 {
+		err = unbindPolicy(client, instanceId, policyId, rmSet)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	if addSet.Len() > 0 {
+		opt := buildPolicyAssociateOpts(instanceId, policyId, utils.ExpandToStringListBySet(addSet))
+		_, err = throttles.Bind(client, opt)
+		if err != nil {
+			return diag.Errorf("error binding policy to the API: %v", err)
+		}
+	}
+
+	return resourceThrottlingPolicyAssociateRead(ctx, d, meta)
+}
+
+func resourceThrottlingPolicyAssociateDelete(_ context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ApigV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating HuaweiCloud APIG v2 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	policyId := d.Get("policy_id").(string)
+	return diag.FromErr(unbindPolicy(client, instanceId, policyId,
+		d.Get("publish_ids").(*schema.Set)))
+}
+
+func resourceThrottlingPolicyAssociateImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import id, must be <instance_id>/<policy_id>")
+	}
+
+	instanceId := parts[0]
+	policyId := parts[1]
+	d.SetId(fmt.Sprintf("%s/%s", instanceId, policyId))
```

Comment:
unnecessary,  pls remove SetId

Reply:
sure, this format and imported are same.